components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngine.kt

```diff
@@ -346,9 +347,22 @@ class GeckoEngine(
             } ?: emptyList()
 
             extensions.forEach { extension ->
+                // https://github.com/mozilla-mobile/android-components/issues/6227 set all installed
+                // extensions to be allowed in private browsing mode. We need to revert back all the
+                // users affected back to default allowPrivateBrowsing, which is false.
+                if (extension.isAllowedInPrivateBrowsing() &&
+                        sharedPref.getBoolean(KEY_DEFAULT_WEB_EXTENSION_ALLOWED_IN_PRIVATE_BROWSING, false)
```

Comment:
This condition doesn't seem right. First time you run this you won't have the key and you default to false, so you won't ever enter the block (see comment below to prevent write on every startup). I think this flag should be "MOZAC_HAS_RESET_WEBEXT_PRIVATE_BROWSING_DEFAULT"...

Reply:
Unresolved this as the change didn't get applied.